community/blog/2022-12 Synchronous secret retrieval in Arcus Security v1.9/README.md

```diff
@@ -0,0 +1,95 @@
+# Synchronous secret retrieval in Arcus Security v1.9
+In some cases, an asynchronous context is missing. This causes a bit of friction that the newest Arcus Security release has fixed with minimal change to your code base.
+
+## The problem with asynchronous-only secret retrieval
+The Arcus secret store is a very reliable and well-used functionality. It provides an alternative to storing sensitive secrets together with public application configuration values ([learn more about why this is important](https://www.codit.eu/blog/introducing-secret-store-net-core/)). Throughout the following releases, the secret store has grown in functionality and usability. It supports caching, secret versions, and lets you retrieve a single secret provider by name and is fully extensible. But there was a piece missing in this structure, and that is synchronous secret retrieval.
+
+When using the secret providers, you can easily use them in asynchronous contexts like an API controller or upon each service method call. The problem happens, though, when you want to register a service in the dependency container. If that service needs to be initialized upon creation, you need the secret at creation time. Dependency containers usually do not work with asynchronous registration ([the one from Microsoft does not](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations)).
+
+The following workaround usually happens when the Arcus secret store is used in such scenarios:
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+using Arcus.Security.Core;
+
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSingleton(serviceProvider =>
+    {
+        var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
+        string secretValue = secretProvider.GetRawSecretAsync("MySecret").GetAwaiter().GetResult();
+
+        return new MyServiceThatRequiresSecretRightAway(secretValue);
+    });
+}
+```
+
+## The solution of sync-support secret providers
+What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that inherits from our main `ISecretProvider` interface. This means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the other way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.
```

Comment:
You're saying here that 'every sync secret provider is also an async secret provider, but before saying that, I think it would be good if we could first say what this ISync... interface is about, without going into too much detail.